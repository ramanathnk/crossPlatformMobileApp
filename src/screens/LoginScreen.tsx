import React, { useState } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableOpacity,
  SafeAreaView,
  ScrollView,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import { StatusBar } from 'expo-status-bar';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import { RootStackParamList } from '../../App';

type LoginScreenNavigationProp = StackNavigationProp<RootStackParamList, 'Login'>;

const LoginScreen: React.FC = () => {
  const navigation = useNavigation<LoginScreenNavigationProp>();
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [showPassword, setShowPassword] = useState(false);

  const handleSignIn = () => {
    // Handle sign in logic here
    console.log('Sign in pressed');
  };

  const handleForgotPassword = () => {
    navigation.navigate('ForgotPassword');
  };

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar style="light" />
      <KeyboardAvoidingView 
        style={styles.keyboardAvoidingView}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      >
        <ScrollView 
          contentContainerStyle={styles.scrollContainer}
          showsVerticalScrollIndicator={false}
          keyboardShouldPersistTaps="handled"
        >
          {/* Logo and App Name */}
          <View style={styles.headerContainer}>
            <View style={styles.logoContainer}>
              <View style={styles.logo}>
                <Text style={styles.logoText}>ðŸ“‹</Text>
              </View>
              <Text style={styles.appName}>SnapTracker</Text>
            </View>
          </View>

          {/* Title Section */}
          <View style={styles.titleSection}>
            <Text style={styles.title}>Admin Registration Portal</Text>
            <Text style={styles.subtitle}>Sign in to manage device registrations</Text>
          </View>

          {/* Form Section */}
          <View style={styles.formContainer}>
            {/* Username Field */}
            <View style={styles.inputContainer}>
              <Text style={styles.inputLabel}>Username</Text>
              <TextInput
                style={styles.textInput}
                placeholder="Enter your username"
                placeholderTextColor="#6B7280"
                value={username}
                onChangeText={setUsername}
                autoCapitalize="none"
                autoCorrect={false}
              />
            </View>

            {/* Password Field */}
            <View style={styles.inputContainer}>
              <Text style={styles.inputLabel}>Password</Text>
              <View style={styles.passwordContainer}>
                <TextInput
                  style={styles.passwordInput}
                  placeholder="Enter your password"
                  placeholderTextColor="#6B7280"
                  value={password}
                  onChangeText={setPassword}
                  secureTextEntry={!showPassword}
                  autoCapitalize="none"
                  autoCorrect={false}
                />
                <TouchableOpacity 
                  style={styles.eyeIcon}
                  onPress={() => setShowPassword(!showPassword)}
                >
                  <View style={styles.eyeIconContainer}>
                    {showPassword ? (
                      // Eye with slash (hidden)
                      <View style={styles.eyeWrapper}>
                        <View style={styles.eyeBase} />
                        <View style={styles.eyePupil} />
                        <View style={styles.eyeSlash} />
                      </View>
                    ) : (
                      // Normal eye (visible)
                      <View style={styles.eyeWrapper}>
                        <View style={styles.eyeBase} />
                        <View style={styles.eyePupil} />
                      </View>
                    )}
                  </View>
                </TouchableOpacity>
              </View>
            </View>

            {/* Sign In Button */}
            <TouchableOpacity style={styles.signInButton} onPress={handleSignIn}>
              <Text style={styles.signInButtonText}>Sign In</Text>
            </TouchableOpacity>

            {/* Forgot Password Link */}
            <TouchableOpacity onPress={handleForgotPassword}>
              <Text style={styles.forgotPasswordText}>Forgot password?</Text>
            </TouchableOpacity>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#1F2937',
  },
  keyboardAvoidingView: {
    flex: 1,
  },
  scrollContainer: {
    flexGrow: 1,
    paddingHorizontal: 24,
    paddingTop: 60,
    paddingBottom: 40,
  },
  headerContainer: {
    alignItems: 'center',
    marginBottom: 40,
  },
  logoContainer: {
    alignItems: 'center',
  },
  logo: {
    width: 60,
    height: 60,
    backgroundColor: '#3B82F6',
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 12,
  },
  logoText: {
    fontSize: 28,
    color: '#FFFFFF',
  },
  appName: {
    fontSize: 24,
    fontWeight: '600',
    color: '#FFFFFF',
    marginTop: 8,
  },
  titleSection: {
    alignItems: 'center',
    marginBottom: 40,
  },
  title: {
    fontSize: 20,
    fontWeight: '600',
    color: '#FFFFFF',
    marginBottom: 8,
    textAlign: 'center',
  },
  subtitle: {
    fontSize: 14,
    color: '#9CA3AF',
    textAlign: 'center',
    lineHeight: 20,
  },
  formContainer: {
    flex: 1,
  },
  inputContainer: {
    marginBottom: 20,
  },
  inputLabel: {
    fontSize: 14,
    color: '#D1D5DB',
    marginBottom: 8,
    fontWeight: '500',
  },
  textInput: {
    backgroundColor: '#374151',
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingVertical: 14,
    fontSize: 16,
    color: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#4B5563',
  },
  passwordContainer: {
    position: 'relative',
  },
  passwordInput: {
    backgroundColor: '#374151',
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingVertical: 14,
    paddingRight: 50,
    fontSize: 16,
    color: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#4B5563',
  },
  eyeIcon: {
    position: 'absolute',
    right: 16,
    top: 14,
    padding: 4,
  },
  eyeIconContainer: {
    width: 20,
    height: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  eyeWrapper: {
    position: 'relative',
    width: 18,
    height: 12,
  },
  eyeBase: {
    width: 18,
    height: 12,
    borderRadius: 9,
    borderWidth: 1.5,
    borderColor: '#9CA3AF',
    backgroundColor: 'transparent',
  },
  eyePupil: {
    position: 'absolute',
    top: 3,
    left: 6,
    width: 6,
    height: 6,
    borderRadius: 3,
    backgroundColor: '#9CA3AF',
  },
  eyeSlash: {
    position: 'absolute',
    top: -2,
    left: -2,
    width: 22,
    height: 1.5,
    backgroundColor: '#9CA3AF',
    transform: [{ rotate: '45deg' }],
  },
  signInButton: {
    backgroundColor: '#3B82F6',
    borderRadius: 8,
    paddingVertical: 16,
    alignItems: 'center',
    marginTop: 8,
    marginBottom: 24,
  },
  signInButtonText: {
    fontSize: 16,
    fontWeight: '600',
    color: '#FFFFFF',
  },
  forgotPasswordText: {
    fontSize: 14,
    color: '#3B82F6',
    textAlign: 'center',
    textDecorationLine: 'underline',
  },
});

export default LoginScreen;
