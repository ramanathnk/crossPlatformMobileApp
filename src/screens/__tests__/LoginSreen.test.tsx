import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react-native'
import '@testing-library/jest-native/extend-expect'
import { NavigationContainer } from '@react-navigation/native'
import LoginScreen from '../LoginScreen'

describe('LoginScreen', () => {
  const setup = () =>
    render(
      <NavigationContainer>
        <LoginScreen />
      </NavigationContainer>
    )

  it('disables Sign In initially', () => {
    const { getByTestId } = setup()
    expect(getByTestId('sign-in-button')).toBeDisabled()
  })

  it.skip('enables Sign In when both fields are filled', async () => {
    const { getByTestId } = setup()

    fireEvent.changeText(getByTestId('username-input'), 'user1')

    try {
      fireEvent.press(getByTestId('toggle-password-visibility'))
      await waitFor(() =>
        expect(getByTestId('password-input').props.secureTextEntry).toBe(false)
      )
    } catch { }


    // Wait for button to be enabled
    await waitFor(() => {
      const btn = getByTestId('sign-in-button')
      expect(btn.props.accessibilityState?.disabled).toBe(false)
    })
  })

  it('matches snapshot', () => {
    const { toJSON } = setup()
    expect(toJSON()).toMatchSnapshot()
  })
})