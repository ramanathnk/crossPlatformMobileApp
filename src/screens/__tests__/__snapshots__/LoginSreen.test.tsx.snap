// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#1F2937",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
          "minHeight": "100%",
          "paddingBottom": 60,
          "paddingHorizontal": 24,
          "paddingTop": 60,
        }
      }
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "marginBottom": 40,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                resizeMode="contain"
                source="FileMock"
                style={
                  {
                    "height": 50,
                    "width": 120,
                  }
                }
              />
            </View>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "600",
                  "marginTop": 8,
                }
              }
            >
              SnapTracker
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "marginBottom": 40,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 20,
                "fontWeight": "600",
                "marginBottom": 8,
                "textAlign": "center",
              }
            }
          >
            Admin Registration Portal
          </Text>
          <Text
            style={
              {
                "color": "#9CA3AF",
                "fontSize": 14,
                "lineHeight": 20,
                "textAlign": "center",
              }
            }
          >
            Sign in to manage device registrations
          </Text>
        </View>
        <View
          style={
            {
              "flex": 0,
              "minHeight": 200,
            }
          }
        >
          <View
            style={
              {
                "marginBottom": 20,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#D1D5DB",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "marginBottom": 8,
                }
              }
            >
              Username
            </Text>
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Enter your username"
              placeholderTextColor="#6B7280"
              style={
                {
                  "backgroundColor": "#374151",
                  "borderColor": "#4B5563",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "paddingHorizontal": 16,
                  "paddingVertical": 14,
                }
              }
              testID="username-input"
              value=""
            />
          </View>
          <View
            style={
              {
                "marginBottom": 20,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#D1D5DB",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "marginBottom": 8,
                }
              }
            >
              Password
            </Text>
            <View
              style={
                {
                  "position": "relative",
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                onChangeText={[Function]}
                placeholder="Enter your password"
                placeholderTextColor="#6B7280"
                secureTextEntry={true}
                style={
                  {
                    "backgroundColor": "#374151",
                    "borderColor": "#4B5563",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "paddingHorizontal": 16,
                    "paddingRight": 50,
                    "paddingVertical": 14,
                  }
                }
                testID="password-input"
                value=""
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "opacity": 1,
                    "padding": 4,
                    "position": "absolute",
                    "right": 16,
                    "top": 14,
                  }
                }
                testID="toggle-password-visibility"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      {
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                >
                  <svg
                    fill="#9CA3AF"
                    height={20}
                    width={20}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#6B7280",
                "borderRadius": 8,
                "marginBottom": 32,
                "marginTop": 8,
                "opacity": 0.6,
                "paddingVertical": 16,
              }
            }
            testID="sign-in-button"
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "fontWeight": "600",
                  },
                  {
                    "color": "#D1D5DB",
                  },
                ]
              }
            >
              Sign In
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#3B82F6",
                  "fontSize": 14,
                  "paddingVertical": 8,
                  "textAlign": "center",
                  "textDecorationLine": "underline",
                }
              }
            >
              Forgot password?
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
